publisher.rb

```diff
@@ -1,33 +1,55 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
+require_relative 'q_it_errors'
 
 class Publisher
   
   attr_reader :sqs, :sleep_period, :queue_name
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
     @queue_name = options[0]
+    @sleep_period = options[1]
+  end
+
+  def self.validate(options)
+    raise QItArgumentError, "Incorrect number of arguments. Expected 2 got #{options.count}" if options.count != 2
+    
+    queue_name = options[0]
+    sleep_period = options[1]
+    valid_queue_name?(queue_name) && valid_sleep_period?(sleep_period)
   end
 
   # For Standard Queue
   def create_std_q
     sqs.create_queue(queue_name: queue_name)
   end
 
-  def valid_sleep_period?(sleep_period)
+  def self.valid_sleep_period?(sleep_period)
     max_sleep_period = /\A([5-9]|1\d{1}|20)\z/
-    !sleep_period.nil? && sleep_period.match(max_sleep_period)
+    
+    raise QItNullSleepTimeError, "QItNullSleepTimeError :: #{self} Sleep period cannot be nil." if sleep_period.nil?
```

Comment:
Not sure we can get a null value as you are checking that you get two values in L17.
As well as checking the number of arguments in app.rb (L41).

Reply:
yeah.. the chances are almost none of getting this kind of error. But I still left that in, just in case. Do you think I should remove it?